app/controllers/whitehall_assets_controller.rb

```diff
@@ -25,4 +29,8 @@ def asset_params
       .require(:asset)
       .permit(:file, :legacy_url_path, :legacy_etag, :legacy_last_modified)
   end
+
+  def existing_asset_with_this_legacy_url_path
```

Comment:
Minor: I'd be inclined to extract this method and the call to `exists?` in the `create` action into a class method on `WhitehallAsset`, e.g. `WhitehallAsset.exists_with_legacy_url_path?(path)`, and then call `if WhitehallAsset.exists_with_legacy_url_path?(asset_params[:legacy_url_path])` from the action.

Reply:
I can't think of a way of doing this without adding two methods, one to find and one to check the existence. I'll leave this change for now and see if I can come up with a refactoring in another PR.